integration/helpers.go

```diff
@@ -571,30 +573,30 @@ type proxyServer struct {
 func (p *proxyServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// validate http connect parameters
 	if r.Method != http.MethodConnect {
-		http.Error(w, fmt.Sprintf("%v not supported", r.Method), http.StatusInternalServerError)
+		trace.WriteError(w, fmt.Errorf("%v not supported", r.Method))
```

Comment:
please don't use fmt.Errorf as it produces not standard HTTP response codes, while `trace.BadParameter` will produce 400 Bad Request and `trace.AccessDenied` will produce 403